repo/plugin.video.nzbdav/resources/lib/newznab_caps.py

```
@@ -75,10 +75,20 @@ def _params(value):
+    if isinstance(xml_text, bytes):
+        probe = xml_text.lower()
+        return b"<!doctype" in probe or b"<!entity" in probe
+    probe = str(xml_text).lower()
+    return "<!doctype" in probe or "<!entity" in probe
+
+
 def parse_caps(xml_text):
     try:
+        if _contains_xml_declaration_markup(xml_text):
+            raise ValueError("DTD and entity declarations are not allowed")
         root = ET.fromstring(xml_text)  # nosec B314 - Python 3.8+ disables entities
```

Ruff (0.15.18)

[error] Using `xml` to parse untrusted data is known to be vulnerable to XML attacks; use `defusedxml` equivalents

(S314)